service/surf/settings/base.py

```diff
@@ -127,6 +128,7 @@
     'django.contrib.auth.middleware.AuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
+    'surf.apps.core.RestrictAdminAccess'
```

Comment:
I think that adding this immediately after the SecurityMiddleware is a good idea. Otherwise untrusted sources would still be able to get some stuff out of our code like session cookies and the entire social auth pipeline. Let's just block that and we don't have to think about possible holes any more.